File: ruby/lib/casilla_impuesto.rb
module Civitas
    class CasillaImpuesto < Casilla
        def initialize(nombre, importe)
            super(nombre)
            @importe=importe
        end
        attr_reader :importe

        def recibeJugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                todos[actual].pagaImpuesto(@importe);
            end
        end
    end
end

File: Civitas_gui/src/civitas/CasillaImpuesto.java
package civitas;
import java.util.ArrayList;

public class CasillaImpuesto extends Casilla {
    private float importe;

    CasillaImpuesto(float cantidad, String nombre) {
        super(nombre);
        importe = cantidad;
    }
    
    @Override
    public String getTipo(){
        return "Impuesto";
    }
    
    @Override
    public float getImporte(){
        return importe;
    }

    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }
}